### ISSUE TITLE
Nothing works .

### ISSUE THREAD

**Issue by gkatsanos** [cmelion/generator-ng2-webpack#69]
I got latest node and npm, I get countless errors, basically all typings related.
The ambientDependencies paths are not found, and when I do run npm run server I get:

```
ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/operator/toPromise.d.ts
(7,59): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/operator/toPromise.d.ts
(7,69): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/operator/toPromise.d.ts
(9,9): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/operator/toPromise.d.ts
(10,26): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/operator/toPromise.d.ts
(10,36): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/observable/PromiseObservable.d.ts
(40,31): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/observable/PromiseObservable.d.ts
(41,26): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/platform/browser/browser_adapter.d.ts
(75,33): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/platform/dom/dom_adapter.d.ts
(85,42): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/platform/browser.d.ts
(78,90): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/http/url_search_params.d.ts
(11,16): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/http/headers.d.ts
(43,59): error TS2304: Cannot find name 'Map'.

ERROR in ./src/app/home/home.ts
(26,12): error TS2304: Cannot find name 'require'.

ERROR in ./src/app/home/home.ts
(28,13): error TS2304: Cannot find name 'require'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/common/directives/ng_class.d.ts
(72,35): error TS2304: Cannot find name 'Set'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/common/pipes/async_pipe.d.ts
(25,38): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/lifecycle/lifecycle_annotations.d.ts
(29,100): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/interfaces.d.ts
(21,107): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/interfaces.d.ts
(39,104): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/interfaces.d.ts
(57,109): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/interfaces.d.ts
(80,109): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/interfaces.d.ts
(102,114): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/directives/router_outlet.d.ts
(27,54): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/directives/router_outlet.d.ts
(33,51): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/directives/router_outlet.d.ts
(38,56): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/directives/router_outlet.d.ts
(47,65): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/directives/router_outlet.d.ts
(58,60): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/instruction.d.ts
(124,34): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/instruction.d.ts
(148,25): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/instruction.d.ts
(164,34): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/instruction.d.ts
(167,25): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/route_definition.d.ts
(20,20): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/route_definition.d.ts
(35,20): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/route_config/route_config_impl.d.ts
(101,19): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/route_registry.d.ts
(52,66): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(56,50): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(68,46): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(86,45): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(99,34): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(107,64): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(112,85): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(120,70): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(128,43): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(132,29): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(138,19): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/router/router.d.ts
(146,70): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/mock/mock_application_ref.d.ts
(16,33): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/compiler/xhr.d.ts
(6,23): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/compiler/xhr_mock.d.ts
(10,23): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/testing/test_component_builder.d.ts
(123,43): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/testing/matchers.d.ts
(4,37): error TS2503: Cannot find namespace 'jasmine'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(1,25): error TS2304: Cannot find name 'MapConstructor'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(2,25): error TS2304: Cannot find name 'SetConstructor'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(4,27): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(4,39): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(7,9): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(8,30): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(11,43): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(12,27): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(14,23): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(15,25): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(100,41): error TS2304: Cannot find name 'Set'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(101,22): error TS2304: Cannot find name 'Set'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/collection.d.ts
(102,25): error TS2304: Cannot find name 'Set'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/dynamic_component_loader.d.ts
(59,148): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/dynamic_component_loader.d.ts
(100,144): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/dynamic_component_loader.d.ts
(105,139): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/dynamic_component_loader.d.ts
(106,135): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/component_resolver.d.ts
(8,53): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/linker/component_resolver.d.ts
(12,44): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/application_ref.d.ts
(38,88): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/application_ref.d.ts
(92,42): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/application_ref.d.ts
(151,33): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/Observable.d.ts
(10,66): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/Observable.d.ts
(66,60): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/rxjs/Observable.d.ts
(66,70): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(2,14): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(8,32): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(9,38): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(10,35): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(10,93): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(11,34): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(11,50): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(12,32): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(12,149): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/promise.d.ts
(13,43): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/async.d.ts
(27,33): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/async.d.ts
(28,45): error TS2304: Cannot find name 'Promise'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/change_detection/differs/default_keyvalue_differ.d.ts
(23,15): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/change_detection/differs/default_keyvalue_differ.d.ts
(25,16): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/di/reflective_provider.d.ts
(103,123): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/core/di/reflective_provider.d.ts
(103,165): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/lang.d.ts
(4,17): error TS2304: Cannot find name 'Map'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/lang.d.ts
(5,17): error TS2304: Cannot find name 'Set'.

ERROR in /Users/gplus/vm/angular2demo/node_modules/angular2/src/facade/lang.d.ts
(71,59): error TS2304: Cannot find name 'Map'.

ERROR in ./src/vendor.ts
(7,22): error TS2304: Cannot find name 'process'.

ERROR in ./src/vendor.ts
(9,3): error TS2304: Cannot find name 'require'.

ERROR in ./src/vendor.ts
(11,3): error TS2304: Cannot find name 'Zone'.

ERROR in ./src/vendor.ts
(11,32): error TS2304: Cannot find name 'require'.

ERROR in ./src/vendor.ts
(14,22): error TS2304: Cannot find name 'process'.

ERROR in ./src/vendor.ts
(17,16): error TS2304: Cannot find name 'require'.

ERROR in ./src/main.ts
(6,27): error TS2305: Module '"/Users/gplus/vm/angular2demo/node_modules/angular2/router"' has no exported member 'LocationStrategy'.

ERROR in ./src/main.ts
(6,45): error TS2305: Module '"/Users/gplus/vm/angular2demo/node_modules/angular2/router"' has no exported member 'HashLocationStrategy'.

ERROR in ./src/main.ts
(20,26): error TS2304: Cannot find name 'process'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/providers/title.spec.ts
(10,1): error TS2304: Cannot find name 'describe'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/providers/title.spec.ts
(13,3): error TS2304: Cannot find name 'beforeEach'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/providers/title.spec.ts
(18,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/directives/x-large.spec.ts
(30,9): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/home.spec.ts
(34,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/home.spec.ts
(38,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/home.spec.ts
(42,5): error TS2304: Cannot find name 'spyOn'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/home.spec.ts
(43,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/home/home.spec.ts
(46,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/src/app/app.spec.ts
(12,1): error TS2304: Cannot find name 'describe'.

ERROR in /Users/gplus/vm/angular2demo/src/app/app.spec.ts
(19,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/test/sanity-test.spec.ts
(9,1): error TS2304: Cannot find name 'describe'.

ERROR in /Users/gplus/vm/angular2demo/test/sanity-test.spec.ts
(11,5): error TS2304: Cannot find name 'expect'.

ERROR in /Users/gplus/vm/angular2demo/test/injector.spec.ts
(11,1): error TS2304: Cannot find name 'describe'.

ERROR in /Users/gplus/vm/angular2demo/test/injector.spec.ts
(13,5): error TS2304: Cannot find name 'expect'.

ERROR in ./src/vendor.ts
Module not found: Error: Cannot resolve module 'zone.js/lib/browser/zone-microtask' in /Users/gplus/vm/angular2demo/src
 @ ./src/vendor.ts 4:0-45

ERROR in ./src/vendor.ts
Module not found: Error: Cannot resolve module 'zone.js/lib/zones/long-stack-trace.js' in /Users/gplus/vm/angular2demo/src
 @ ./src/vendor.ts 8:33-81

ERROR in ./src/vendor.ts
Module not found: Error: Cannot resolve module 'github-api/github' in /Users/gplus/vm/angular2demo/src
 @ ./src/vendor.ts 20:0-28
```

**Comment by cmelion** [comment]
Looks like a proxying issue:
see https://github.com/npm/npm/issues/2119 for possible remedies.

**Comment by gkatsanos** [comment]
Could you re-open? I am at home with a normal standard no-proxy connection. the actual paths in typings.json were 404 links (I visited them directly)
I changed generators and things work out - of - the - box there. Thanks for making this but please make it usable?

**Comment by cmelion** [comment]
I'm not clear on what is happening, you mention "run npm run server" which would simply launch json-server in an ng2-webpack generated project.

Also your error logs reference "node_modules/angular2", this project uses node_modules/@angular.

This project has a docker image that is rebuilt each time changes are made in master, I confirmed that it builds and runs as expected within the docker image, including tests.

Are you sure you have the right generator?

**Comment by gkatsanos** [comment]
The very first error I got was that post install couldn't finish because of
typings. After trying to solve this I tried different things that maybe
didn't help.
I will recreate an empty project and log all console errors soon.
My platform btw is a latest osx, latest node (6.x) and latest npm.

One of my questions was about global packages. Should I install typescript
and typings globally or locally?

Double checking the project and following up with the error log.

Many thanks for your replies!
